workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/Readers/BlittableReaderBase.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.IO;
+using Improbable.Gdk.Core.MonoBehaviours;
+using Unity.Entities;
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    internal abstract class BlittableReaderBase<TSpatialComponentData, TComponentUpdate>
+        : ReaderBase<TSpatialComponentData, TComponentUpdate>
+        where TSpatialComponentData : struct, ISpatialComponentData, IComponentData
+        where TComponentUpdate : ISpatialComponentUpdate
+    {
+        protected BlittableReaderBase(Entity entity, EntityManager entityManager, ILogDispatcher logDispatcher) : base(entity, entityManager, logDispatcher)
+        {
+        }
+
+        public sealed override TSpatialComponentData Data
+        {
+            get
+            {
+                try
+                {
+                    return EntityManager.GetComponentData<TSpatialComponentData>(Entity);
```

Comment:
I wonder how fast/slow it is to always query the data like this? Not that I see any other option though.

Reply:
I think it won't be a bottleneck for a while